### PR TITLE
fix misplaced laszip tiles

### PR DESCRIPTION
**Pull request by NEDJIMAbelgacem**
## Description
This PR fixes the bug mentioned in https://github.com/wonder-sk/point-cloud-experiments/issues/75 .
Where the offset/scale values need to be taken from the .laz file metadata.

### CODE REVIEW

File: src/3d/symbols/qgspointcloud3dsymbol_p.cpp [hunk 1]
```diff
@@ -270,9 +274,9 @@ void QgsSingleColorPointCloud3DSymbolHandler::processNode( QgsPointCloudIndex *p
     qint32 iy = *( qint32 * )( ptr + i * recordSize + 4 );
     qint32 iz = *( qint32 * )( ptr + i * recordSize + 8 );
 
-    double x = offset.x() + scale.x() * ix;
-    double y = offset.y() + scale.y() * iy;
-    double z = ( offset.z() + scale.z() * iz ) * zValueScale + zValueOffset;
+    double x = blockOffset.x() + blockScale.x() * ix;
```
Comment (pblottiere) [inline src/3d/symbols/qgspointcloud3dsymbol_p.cpp:277]:
```suggestion
    const double x = blockOffset.x() + blockScale.x() * ix;
```

File: src/3d/symbols/qgspointcloud3dsymbol_p.cpp [hunk 2]
```diff
@@ -270,9 +274,9 @@ void QgsSingleColorPointCloud3DSymbolHandler::processNode( QgsPointCloudIndex *p
...
+    double y = blockOffset.y() + blockScale.y() * iy;
```
Comment (pblottiere) [inline src/3d/symbols/qgspointcloud3dsymbol_p.cpp:278]:
```suggestion
    const double y = blockOffset.y() + blockScale.y() * iy;
```

File: src/3d/symbols/qgspointcloud3dsymbol_p.cpp [hunk 3]
```diff
@@ -270,9 +274,9 @@ void QgsSingleColorPointCloud3DSymbolHandler::processNode( QgsPointCloudIndex *p
...
+    double z = ( blockOffset.z() + blockScale.z() * iz ) * zValueScale + zValueOffset;
```
Comment (pblottiere) [inline src/3d/symbols/qgspointcloud3dsymbol_p.cpp:279]:
```suggestion
    const double z = ( blockOffset.z() + blockScale.z() * iz ) * zValueScale + zValueOffset;
```

File: src/3d/symbols/qgspointcloud3dsymbol_p.cpp (hunk 1 above)
Comment (pblottiere) [reply src/3d/symbols/qgspointcloud3dsymbol_p.cpp:277]: Ooops, it's used in `transformInPlace` later.

File: src/core/pointcloud/qgspointcloudrenderer.h [hunk 4]
```diff
@@ -79,11 +79,21 @@ class CORE_EXPORT QgsPointCloudRenderContext
      */
     QgsVector3D scale() const { return mScale; }
 
+    /**
+     * Sets the scale of the layer's int32 coordinates compared to CRS coords.
+     */
```
Comment (pblottiere) [inline src/core/pointcloud/qgspointcloudrenderer.h:84]: I think that `\since` is missing.

File: src/core/pointcloud/qgspointcloudrenderer.h [hunk 5]
```diff
@@ -79,11 +79,21 @@ class CORE_EXPORT QgsPointCloudRenderContext
...
+    void setScale( QgsVector3D scale ) { mScale = scale; }
```
Comment (pblottiere) [inline src/core/pointcloud/qgspointcloudrenderer.h:85]: const reference `scale`?

File: src/core/pointcloud/qgspointcloudrenderer.h [hunk 6]
```diff
@@ -79,11 +79,21 @@ class CORE_EXPORT QgsPointCloudRenderContext
...
+
     /**
      * Returns the offset of the layer's int32 coordinates compared to CRS coords.
      */
     QgsVector3D offset() const { return mOffset; }
 
+    /**
+     * Sets the offset of the layer's int32 coordinates compared to CRS coords.
```
Comment (pblottiere) [inline src/core/pointcloud/qgspointcloudrenderer.h:93]: I think that `\since` is missing.

File: src/core/pointcloud/qgspointcloudrenderer.h [hunk 7]
```diff
@@ -79,11 +79,21 @@ class CORE_EXPORT QgsPointCloudRenderContext
...
+     */
+    void setOffset( QgsVector3D offset ) { mOffset = offset; }
```
Comment (pblottiere) [inline src/core/pointcloud/qgspointcloudrenderer.h:95]: const reference offset?

**Comment by pblottiere** [comment]
Hello @NEDJIMAbelgacem,

I just took a quick look on code style. A proper review is needed.

**Comment by NEDJIMAbelgacem** [comment]
> Hello @NEDJIMAbelgacem,
> 
> I just took a quick look on code style. A proper review is needed.

I addressed your reviews. Thanks!

**Comment by wonder-sk** [comment]
hi @NEDJIMAbelgacem I would suggest that we do not introduce a new class (QgsCustomPointCloudBlock) and rather make offset(), scale() as members of QgsPointCloudBlock. The logic on the client side then would be to always use offset+scale from the returned block. That should make things simpler and less error prone...

File: src/core/pointcloud/qgseptdecoder.cpp [hunk 8]
```diff
@@ -468,30 +471,32 @@ QgsPointCloudBlock *__decompressLaz( FileType &file, const QgsPointCloudAttribut
...
-           count,
-           requestedAttributes,
-           data
-         );
+  QgsPointCloudBlock *block = new QgsPointCloudBlock(
+    count,
+    requestedAttributes,
+    data, scale, offset
+  );
+  return block;
```
Comment (pblottiere) [inline src/core/pointcloud/qgseptdecoder.cpp:479]: Why do we create a new temporary variable `block` instead of just returning the new instance?

**Review by wonder-sk — APPROVED** [review]

**Pull request merged**